slides/testing/validator/duplicate-id-validator.ts

```diff
@@ -0,0 +1,61 @@
+import { FileParser } from './../parser/file-parser';
+import { IValidator } from './validator.interface';
+import { File } from './../parser/file';
+import * as fs from 'fs';
+import * as path from 'path';
+
+export class DuplicateIdValidator implements IValidator {
+
+  private duplicateSlides: any;
+  private basePath = path.resolve(__dirname, '../../../src/content/');
+  
+  get duplicates() {
+    return this.duplicateSlides;
+  }
+
+  validate(): boolean {
+    let files = this.getFiles();
+    let slideIds = this.getSlideIds(files);
+    this.duplicateSlides  = this.getDuplicates(slideIds);
+    return this.duplicateSlides.size === 0 ? true : false;
+  }
```

Comment:
We don't actually seem to be using the boolean returned from the validate method at all here - you check for `duplicates.size === 0`, in which case, storing the duplicates as internal state is kind of meaningless. You could just have validate return the duplicates themselves and make this class stateless since aside from `getFiles`, your methods are pure anyway.

It also appears to me that `this.getDuplicates(slideIds)` is returning an array rather than a Set